evidence/pool.go

```diff
@@ -58,12 +64,13 @@ func NewPool(evidenceDB dbm.DB, stateDB sm.Store, blockStore BlockStore) (*Pool,
 	}
 
 	pool := &Pool{
-		stateDB:       stateDB,
-		blockStore:    blockStore,
-		state:         state,
-		logger:        log.NewNopLogger(),
-		evidenceStore: evidenceDB,
-		evidenceList:  clist.New(),
+		stateDB:           stateDB,
+		blockStore:        blockStore,
+		state:             state,
+		logger:            log.NewNopLogger(),
+		evidenceStore:     evidenceDB,
+		evidenceList:      clist.New(),
+		consensusEvBuffer: make([]types.Evidence, 0),
```

Comment:
Do we need to bound these somehow? If other nodes can send us an arbitrary amount of evidence then they can potentially OOM the node.

Reply:
This is created in the consensus engine and not gossiped by other nodes so it can't necessarily be arbitrary but if a validator did double vote say 1000 times then we would keep that. 

Also important to note that the amount of possible times a single validator can double vote is restricted by the round number. In the first round there can be only 1 double vote (one for nil and one for the block). In the second round we can introduce a new block ID and thus we have 3 permutations. In round 10 we can have 55. But I'm not even sure if the other nodes store all the possible blocks proposed.

It was also discussed about having a map[valAddress & height] Evidence that way only a single evidence per validator per height is stored but I don't think this is implemented yet. It would be relatively easy to do if we wanted to

Reply:
That makes sense, thanks for clarifying - this is fine.